canvas.cpp

```diff
@@ -976,9 +976,14 @@ void Canvas::setGrid(bool newGrid)
 	grid = newGrid;
 }
 
-void Canvas::checkAllTracksInMapConnected()
+bool Canvas::checkAllTracksInMapConnected()
 {
-	map->checkAllTracksConnected();
+	return map->checkAllTracksConnected();
+}
+
+int Canvas::getTrackTotal()
+{
+	return map->getTotalTrack();
 }
 
 void Canvas::clickCreateAddElement(ElementChosen elementToAdd, int overallX, int overallY)
```

Comment:
I am not a C++ maestro but would it not be better to pass by reference here, i.e. `ElementChosen&`? Depends how you are using the object. Also if the object is not being altered consider `const ElementChosen&`.